packages/modeling/src/colors/colorize.d.ts

```diff
@@ -1,9 +1,12 @@
-import { Geometry } from '../geometries/types'
+import { Colored, Geometry } from '../geometries/types'
 import RecursiveArray from '../utils/recursiveArray'
 
 import { RGB, RGBA } from './types'
 
 export default colorize
 
 declare function colorize<T extends Geometry>(color: RGB | RGBA, object: T): T
-declare function colorize(color: RGB | RGBA, ...objects: RecursiveArray<Geometry>): Array<Geometry>
+declare function colorize<T>(color: RGB | RGBA, object: T): T & Colored
+
+declare function colorize<T extends RecursiveArray<Geometry>>(color: RGB | RGBA, ...objects: T): T
```

Comment:
Why is this declaration necessary? I don't understand the need to have 'T extends RecursiveArray<Geometry>'

Reply:
This definition is not necessary and is wrong, so I'll fix it. And the return value is a flat array.